### PR TITLE
Added new DTLS-SRTP protection profiles from RFC 7714

### PR DESCRIPTION
**Pull request by ukreator**
RFC 7714 was published recently with new SRTP profiles. This simple PR adds it to the existing DTLS-SRTP implementation.

### CODE REVIEW

**Comment by kroeckx** [comment]
I'm not sure what this commit gets you, it doesn't seem to implement anything from that RFC.

**Comment by mattcaswell** [comment]
It's just adding the protection profiles from section 14.2 (which sounds like a fairly reasonable request)

**Comment by kroeckx** [comment]
I guess I just have no idea how you use DTLS-SRTP with OpenSSL.  I was expecting that those other values from that RFC should be used somewhere, but maybe that's not in the OpenSSL code?

**Comment by mattcaswell** [comment]
Exactly. You use SSL_CTX_set_tlsext_use_srtp(), SSL_set_tlsext_use_srtp(), SSL_get_srtp_profiles() and SSL_get_selected_srtp_profile() to get and set the profiles that are then used within the SRTP connection (outside of OpenSSL).

File: include/openssl/srtp.h [hunk 1]
```diff
@@ -130,6 +130,10 @@ extern "C" {
 # define SRTP_NULL_SHA1_80      0x0005
 # define SRTP_NULL_SHA1_32      0x0006
 
+/* AEAD SRTP protection profiles from RFC 7714 */
+# define SRTP_AEAD_AES_128_GCM 0x0007
```
Comment (mattcaswell) [inline include/openssl/srtp.h]: minor nit: could we make the values line up with the definitions above?

**Comment by kroeckx** [comment]
So does that mean that this code outside OpenSSL can not use those profiles just because OpenSSL didn't add this profile number it doesn't do anything with?  Or is there some other way for the application to set those profiles so that OpenSSL doesn't always need to get updated?  This all looks like some "custom extension" thing.

**Comment by mattcaswell** [comment]
Yes. Applications can't negotiation the profiles if OpenSSL doesn't know about them.

**Comment by ukreator** [comment]
Fixed formatting. @mattcaswell, should I recreate the PR to contain one commit only?

**Comment by mattcaswell** [comment]
Just squash the commits and push -f to keep it within this PR.

**Comment by ukreator** [comment]
@mattcaswell, done

**Comment by mattcaswell** [comment]
+1

**Comment by mattcaswell** [comment]
We need a second team review now.

**Comment by kroeckx** [comment]
+1

**Comment by mattcaswell** [comment]
Pushed to master. Thanks for the contribution.

**Pull request closed without merging**